util/netcdf_generator.py

```diff
@@ -86,9 +86,13 @@ def _filter_params(self, ds, params):
             params_to_filter.append('annotations')
 
         for key in ds.data_vars:
-            if key not in params_to_filter:
+            # do not remove qc parameters '%s_qc_executed' and '%s_qc_results'
+            if key not in params_to_filter and not self._is_qc_parameter(key):
                 ds = ds.drop(key)
         return ds
+        
+    def _is_qc_parameter(self, param):
```

Comment:
suggest moving comment from line 89 down to this function to help developers understand that you are doing a wildcard search on the parameter name.

Reply:
A comment for the function is preferred (PEP-8), but there is no need to explain the syntax of the language.